### PR TITLE
HWKAPM-539 REST endpoint for retrieving trace completion times

### PR DESCRIPTION
**Pull request by objectiser**

### CODE REVIEW

**Comment by objectiser** [comment]
@pavolloffay Would you be able to review this when you get a chance? This PR just enables the trace completion times to be directly exposed via a REST endpoint.

File: server/rest/src/main/java/org/hawkular/apm/server/rest/AnalyticsHandler.java [hunk 1]
```diff
@@ -500,6 +501,108 @@ public void getTraceCompletionFaultCount(
...
+    @POST
+    @Path("trace/completion/times")
+    @Produces(APPLICATION_JSON)
+    @ApiOperation(
+            value = "Get the trace completion times associated with criteria",
+            response = List.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Success"),
+            @ApiResponse(code = 500, message = "Internal server error") })
+    public void getTraceCompletionTimes(
```
Comment (pavolloffay) [inline server/rest/src/main/java/org/hawkular/apm/server/rest/AnalyticsHandler.java]: Is't the `@GET @Path("trace/completion/times")` doing exactly the same? Or is there any issue with passing some parameters of the `Criteria` as query params?

File: server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/AnalyticsServiceElasticsearch.java [hunk 2]
```diff
@@ -282,6 +282,57 @@ public long getTraceCompletionFaultCount(String tenantId, Criteria criteria) {
...
+                return null;
+            }
+
+            List<CompletionTime> ret = new ArrayList<CompletionTime>();
+
+            for (SearchHit searchHitFields : response.getHits()) {
+                try {
+                    ret.add(mapper.readValue(searchHitFields.getSourceAsString(),
+                            CompletionTime.class));
+                } catch (Exception e) {
```
Comment (pavolloffay) [inline server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/AnalyticsServiceElasticsearch.java]: Isn't it better to catch more concrete exception (`JsonParseException`)?

File: server/rest/src/main/java/org/hawkular/apm/server/rest/AnalyticsHandler.java (hunk 1 above)
Comment (objectiser) [inline server/rest/src/main/java/org/hawkular/apm/server/rest/AnalyticsHandler.java]: Yes it is doing the same - however for convenience I provide two ways to invoke the operation, one using GET and passing the parameters, but from a Javascript app (i.e. the UI) it is better to POST an object representation of the Criteria being managed in the javascript code.

File: server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/AnalyticsServiceElasticsearch.java (hunk 2 above)
Comment (objectiser) [inline server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/AnalyticsServiceElasticsearch.java]: Guilty :) its just me being lazy - I'll fix.

**Comment by objectiser** [comment]
@pavolloffay Thanks!

**Pull request merged**